crates/topos-tce-broadcast/src/lib.rs

```diff
@@ -82,13 +82,17 @@ pub enum DoubleEchoCommand {
     /// When echo reply received
     Echo {
         from_peer: PeerId,
+        validator_id: ValidatorId,
         certificate_id: CertificateId,
+        signature: Signature,
     },
 
     /// When ready reply received
     Ready {
         from_peer: PeerId,
+        validator_id: ValidatorId,
         certificate_id: CertificateId,
+        signature: Signature,
     },
```

Comment:
Same question as elsewhere: do you think we should have types for this data so we never accidentally build an `Echo` with data from a `Ready`?

Reply:
@dvdplm The serialization makes it near-impossible, but we'll have to refactor the message system at some point.

Reply:
Relying on protobuff types and use Envelope to gather metadata and payload. Didn't design this in detail yet.